src/engine/matching-engine.ts

```diff
@@ -0,0 +1,166 @@
+import type { Order } from "schemas";
+import { markets } from "../constants";
+import { checkSignatureOfOrder } from "../signing";
+import type { Market, MarketId } from "../types";
+import { zeroAddress } from "viem";
+import { randomBytes } from "node:crypto";
+import { HTTPError } from "../utils";
+
+type Hex = `0x${string}`;
+
+type LimitOrder = { id: string; signature: Hex; user: Hex; order: Order; timestamp?: bigint };
+
+type LimitOrderRaw = { signature: Hex; user: Hex; order: Order };
```

Comment:
i forgot to add this, it might be much cleaner
```suggestion
type LimitOrder = LimitOrderRaw & { id: string; timestamp?: bigint };

type LimitOrderRaw = { signature: Hex; user: Hex; order: Order };
```